extensions/ql-vscode/src/variant-analysis/code-scanning-pack.ts

```diff
@@ -25,20 +28,30 @@ export async function resolveCodeScanningQueryPack(
 
   // Resolve queries
   void logger.log(`Resolving queries for pack: ${packName}`);
-  const suitePath = join(
-    packDir,
-    "codeql-suites",
-    `${language}-code-scanning.qls`,
-  );
+
+  const suiteFile = await file({
+    postfix: ".qls",
+  });
+  const suitePath = suiteFile.path;
+  const suiteYaml: SuiteInstruction[] = [
+    {
+      import: `codeql-suites/${language}-code-scanning.qls`,
+      from: `${downloadedPack.name}@${downloadedPack.version}`,
+    },
+    // Exclude any non-problem queries
+    {
+      exclude: {
+        kind: ["diagnostic", "metric"],
+      },
```

Comment:
Would
```
include: {
  kind: SARIF_RESULTS_QUERY_KINDS,
}
```
have the same effect? Or maybe include both filters?

The original intention here was that we limit to only queries that can produce SARIF because we don't want to have to deal with displaying raw results from multiple queries right now. That'll come later once we move multi-query MRVA out of the prototype phase.

Reply:
I tested this and unfortunately this doesn't have any effect. After including an existing suite, you can only override it by using `exclude`, not by using `include`. `include` would simply add all queries that match that `kind`, but wouldn't exclude any of the existing already included queries.

Reply:
It's buried in the documentation at the end of [Reusability Examples](https://docs.github.com/en/code-security/codeql-cli/using-the-advanced-functionality-of-the-codeql-cli/creating-codeql-query-suites#reusability-examples) but I believe we can follow this example:
```yaml
- import: codeql-suites/cpp-security-and-quality.qls
  from: codeql/cpp-queries
- exclude: {}
- include:
    precision:
      - very-high
      - high
```

> Notice the empty `exclude` instruction. This is required to ensure that the subsequent `include` instruction is able to filter queries from the imported suite.

I tried it out locally with the following suite file and it seems to behave as intended.

Reply:
I don't believe it will cause it to include queries that weren't in the original code scanning suite. That would be the difference between `import` and `apply`.

I tried to test this. The following suite contains 75 queries:
```yaml
- import: codeql-suites/java-code-scanning.qls
  from: codeql/java-queries@0.8.9
```

If I exclude `diagnostic` and `metric` queries then it decreases to 62 queries:
```yaml
- import: codeql-suites/java-code-scanning.qls
  from: codeql/java-queries@0.8.9
- exclude:
    kind:
      - diagnostic
      - metric
```

If I instead exclude all queries and then only import `problem` and `path-problem` queries then we get the exact same 62 queries:
```yaml
- import: codeql-suites/java-code-scanning.qls
  from: codeql/java-queries@0.8.9
- exclude: {}
- include:
    kind:
      - problem
      - path-problem
```

And to show the difference vs using `apply`, if I change `import` to `apply` in the last case then it sky rockets to 453 queries:
```yaml
- apply: codeql-suites/java-code-scanning.qls
  from: codeql/java-queries@0.8.9
- exclude: {}
- include:
    kind:
      - problem
      - path-problem
```

---

However, this likely doesn't matter and right now we could choose either option. As shown they give the exact same result because we know that the code scanning suites will only contain problem queries and diagnostic/metric queries.

My (possibly incorrect) understanding of the future for this feature is that eventually (no timeline yet) we'll implement viewing raw results and we'll likely also want to expand it to run on custom suites as well as the code scanning suites. For the code scanning suites we'll still want to exclude the metric/diagnostic queries as they're likely not what the user wants. For custom suites if we haven't implemented raw results yet then we'll want to limit it to problem queries, but it's also likely that implementing raw results will come first so it won't matter by that point anyway.

In short, I think including the kinds we want is still a bit cleaner than excluding the ones we don't, but for the code scanning suites they're equivalent and should remain that way for the foreseeable future. If you like the current approach and want to keep it as it is that's fine.

Reply:
Thanks, I didn't realize that when specifying an empty `exclude`, the other instructions from the imported file still apply. I agree that only including the query kinds we support makes more sense, so I'll change this to use the empty `exclude` followed by an `include` with `SARIF_RESULTS_QUERY_KINDS`.